tendrl/node_agent/monitoring/collectd/collectors/gluster/heavy_weight/tendrl_glusterfs_profile_info.py

```diff
@@ -359,130 +355,110 @@ def get_volume_profile_metrics(CLUSTER_TOPOLOGY, CONFIG):
             ] = inode_hits
             t_name = "clusters.%s.volumes.%s.nodes.%s.bricks.%s." \
                 "entry_ops"
-            ret_val[
+            self.profile_info[
                 t_name % (
-                    CONFIG['integration_id'],
+                    self.CONFIG['integration_id'],
                     volName,
                     brick_host.replace('.', '_'),
                     brickName.split(':')[1].replace('/', '|')
                 )
             ] = entry_hits
             t_name = "clusters.%s.nodes.%s.bricks.%s." \
                 "read_write_ops"
-            ret_val[
+            self.profile_info[
                 t_name % (
-                    CONFIG['integration_id'],
+                    self.CONFIG['integration_id'],
                     brick_host.replace('.', '_'),
                     brickName.split(':')[1].replace('/', '|')
                 )
             ] = read_write_hits
             t_name = "clusters.%s.nodes.%s.bricks.%s." \
                 "lock_ops"
-            ret_val[
+            self.profile_info[
                 t_name % (
-                    CONFIG['integration_id'],
+                    self.CONFIG['integration_id'],
                     brick_host.replace('.', '_'),
                     brickName.split(':')[1].replace('/', '|')
                 )
             ] = lock_hits
             t_name = "clusters.%s.nodes.%s.bricks.%s." \
                 "inode_ops"
-            ret_val[
+            self.profile_info[
                 t_name % (
-                    CONFIG['integration_id'],
+                    self.CONFIG['integration_id'],
                     brick_host.replace('.', '_'),
                     brickName.split(':')[1].replace('/', '|')
                 )
             ] = inode_hits
             t_name = "clusters.%s.nodes.%s.bricks.%s." \
                 "entry_ops"
-            ret_val[
+            self.profile_info[
                 t_name % (
-                    CONFIG['integration_id'],
+                    self.CONFIG['integration_id'],
                     brick_host.replace('.', '_'),
                     brickName.split(':')[1].replace('/', '|')
                 )
             ] = entry_hits
-    return ret_val
-
-
-def get_metrics(CLUSTER_TOPOLOGY, CONFIG):
-    profile_info = {}
-    heal_stats = {}
-    profile_info = get_volume_profile_metrics(
-        CLUSTER_TOPOLOGY,
-        CONFIG
-    )
-    heal_stats = tendrl_gluster_heal_info.get_metrics(
-        CLUSTER_TOPOLOGY,
-        CONFIG
-    )
-    profile_info.update(heal_stats)
-    return profile_info
 
+    def get_volume_profile_metrics(self):
+        global READ_WRITE_OPS
+        global LOCK_OPS
+        global INODE_OPS
+        global ENTRY_OPS
+        self.profile_info = {}
+        volumes = self.CLUSTER_TOPOLOGY.get('volumes', [])
+        # threads = []
+        for volume in volumes:
+            self.process_volume_profile_info(volume)
+            # thread = threading.Thread(
```

Comment:
Commented lines

Reply:
Same as in https://github.com/Tendrl/node-agent/pull/599/files#r136752747